utils/big_decimal.cc

```diff
@@ -143,6 +143,99 @@ std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
     return x.compare(y) <=> 0;
 }
 
+std::strong_ordering big_decimal::operator<=>(const big_decimal& other) const
+{
+    if (_scale == other._scale) {
+        return _unscaled_value.compare(other._unscaled_value) <=> 0;
+    }
+
+    const int sign = boost::multiprecision::sign(_unscaled_value) < 0 ? -1 : 1;
+    const int sign_other = boost::multiprecision::sign(other._unscaled_value) < 0 ? -1 : 1;
```

Comment:
Why coerce sign into -1/+1? if both signs equal zero, we don't need to compare anything, and if one is zero and the other is nonzero, the comparison below works. With the coercion, you lose the fast path when comparing 0 and some large positive number.

Reply:
I couldn't find any docs on this, so I was afraid to make assumptions.

Reply:
I removed the coercion in v4.